examples/grpc-server-with-prometheus/server/server.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	pb "github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/protobuf"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// DemoServiceServer defines a Server.
+type DemoServiceServer struct{}
+
+func newDemoServer() *DemoServiceServer {
+	return &DemoServiceServer{}
+}
+
+// SayHello implements a interface defined by protobuf.
+func (s *DemoServiceServer) SayHello(ctx context.Context, request *pb.HelloRequest) (*pb.HelloResponse, error) {
+	pushCounter.Inc()
```

Comment:
can we add some labels "just for fun" to show how to inject them? In that case, you can also show how to init the counter finally (:  (Without labels it is fine - with labels the fun begins)

Reply:
Sorry guys, I don't understand your suggestions. what is label "just for fun"?
If you can , please give me a distinct example.

Reply:
No worries.

I mean to define some dynamic labels to pushCounter metric to make it more realistic. We can for example add "name" label for fun:
```
pushCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
		Name: "repository_pushes",
		Help: "Number of pushes to external repository.",
}, []string{"name"})
```

And then you can incement specific labeled metric by
`pushCounter.WithLabelValues(request.Name).Inc()`

Now we also can explain that user can hit the issue of not initialised metric (what @mwitkow suggested to show). Basically if you register this metric and no SayHello will be invoked. You will see no `repository_pushes` on metric page.

After first call with name `abc` you will see  `respositoty_pushes{name=abc} 1` 
As you can see we missed 0 value, so it can look weird in certain cases (e.g on dashboard)

To make sure the metric is intialized for certain name we can do just `pushCounter.WithLabelValues("abc")` in the beginning. If you have label that you know upfront what values it will have - it makes sense to init them. Name label is not the best example, but we can say that we expect `test` name (because you use it in your client code), so we can init this metric with this value. Does it explain this @fengzixu?

Do I miss anything here @brancz ?